packages/blocks/src/root-block/widgets/linked-doc/index.ts

```diff
@@ -19,72 +19,27 @@ import {
 } from '../../../_common/utils/query.js';
 import { getCurrentNativeRange } from '../../../_common/utils/selection.js';
 import { getPopperPosition } from '../../../root-block/utils/position.js';
-import { type LinkedDocOptions, getMenus } from './config.js';
+import { type LinkedMenuGroup, getMenus } from './config.js';
 import { LinkedDocPopover } from './linked-doc-popover.js';
 
-export function showLinkedDocPopover({
-  editorHost,
-  inlineEditor,
-  range,
-  container = document.body,
-  abortController = new AbortController(),
-  options,
-  triggerKey,
-}: {
-  editorHost: EditorHost;
-  inlineEditor: AffineInlineEditor;
-  range: Range;
-  container?: HTMLElement;
-  abortController?: AbortController;
-  options: LinkedDocOptions;
-  triggerKey: string;
-}) {
-  const disposables = new DisposableGroup();
-  abortController.signal.addEventListener('abort', () => disposables.dispose());
-
-  const linkedDoc = new LinkedDocPopover(
-    editorHost,
-    inlineEditor,
-    abortController
-  );
-  linkedDoc.options = options;
-  linkedDoc.triggerKey = triggerKey;
-  // Mount
-  container.append(linkedDoc);
-  disposables.add(() => linkedDoc.remove());
-
-  // Handle position
-  const updatePosition = throttle(() => {
-    const linkedDocElement = linkedDoc.linkedDocElement;
-    assertExists(
-      linkedDocElement,
-      'You should render the linked doc node even if no position'
-    );
-    const position = getPopperPosition(linkedDocElement, range);
-    linkedDoc.updatePosition(position);
-  }, 10);
-  disposables.addFromEvent(window, 'resize', updatePosition);
-  const scrollContainer = getViewportElement(editorHost);
-  if (scrollContainer) {
-    // Note: in edgeless mode, the scroll container is not exist!
-    disposables.addFromEvent(scrollContainer, 'scroll', updatePosition, {
-      passive: true,
-    });
-  }
-
-  // Wait for node to be mounted
-  setTimeout(updatePosition);
-
-  disposables.addFromEvent(window, 'mousedown', (e: Event) => {
-    if (e.target === linkedDoc) return;
-    abortController.abort();
-  });
+export const AFFINE_LINKED_DOC_WIDGET = 'affine-linked-doc-widget';
 
-  return linkedDoc;
+export interface LinkedWidgetConfig {
+  // The first item of the trigger keys will be the primary key
```

Comment:
Just a hint: using the `/* */` style comments for interface fields would allow better intellisense in VSCode 😀

Reply:
Cool. Code annotation have been updated.